### PR TITLE
fix(init): fixes an issue where the init command exited with bad credentials

### LINKED ISSUE #3488: Github credentials have expired. CLI throwing error but not prompting for updated credentials.
**Describe the bug**

When I try and init a new project using the netlify-cli, everything works as planned until the cli attempts to link the netlify project to the associated git repository. At that point it throws the following error:

**Error: Failed retrieving GitHub repository information with error: Bad credentials**

I've looked through the CLI docs but I'm not sure how to reset my Github credentials.

**To Reproduce**

For me, this occurs during the `netlify init` process with a new netlify project and an existing (linked) github repository. Again, the netlify project is created but the `init` process ends in error and I must visit the netlify.com interface to link the Github project.

**Expected behavior**

I suppose I expected the CLI to prompt me to update my Github credentials or for the documentation to explain how I can do this.

**CLI Output**

```
Error: Failed retrieving GitHub repository information with error: Bad credentials
```

### PR DESCRIPTION
**Pull request by lukasholzer**
🎉 Thanks for submitting a pull request! 🎉

#### Summary

Fixes #3488

Fixes an issue where the init command failed with bad credentials error if the github token was not valid:

CleanShot 2021-11-12 at 14 56 11

With the newly introduced changes, the validity check of the token does not only check if there is any value it checks it by requesting the GitHub API.

If GitHub returns an error we generate now new credentials like it was never connected:

CleanShot 2021-11-12 at 14 57 32

---

For us to review and ship your PR efficiently, please perform the following steps:

- [x] Open a [bug/issue](https://github.com/netlify/cli/issues/new/choose) before writing your code 🧑‍💻. This ensures we can discuss the changes and get feedback from everyone that should be involved. If you\`re fixing a typo or something that\`s on fire 🔥 (e.g. incident related), you can skip this step.
- [x] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.
- [x] Update or add tests (if any source code was changed or added) 🧪
- [x] Update or add documentation (if features were changed or added) 📝
- [x] Make sure the status checks below are successful ✅

**A picture of a cute animal (not mandatory, but encouraged)**

### CODE REVIEW

File: src/utils/gh-auth.js [hunk 1]
```diff
@@ -108,15 +113,19 @@ const authWithToken = async () => {
...
+ * Get a github token
+ * @returns {Promise<Token>} Returns a Promise with a token object
+ */
 const getGitHubToken = async () => {
   log('')
 
   const withNetlify = await promptForAuthMethod()
   if (withNetlify) {
-    return await authWithNetlify()
+    return authWithNetlify()
```
Comment (erezrokah) [inline src/utils/gh-auth.js:125]: See [here](https://github.com/goldbergyoni/nodebestpractices#-212-always-await-promises-before-returning-to-avoid-a-partial-stacktrace) on the reason we use `return await`

File: src/utils/init/config-github.js [hunk 2]
```diff
@@ -6,34 +6,55 @@ const { getGitHubToken: ghauth } = require('../gh-auth')
...
-  if (isValidToken(githubToken)) {
-    return githubToken.token
+  if (githubToken && githubToken.user && githubToken.token) {
+    try {
+      const octokit = getGitHubClient(githubToken.token)
+      const { status } = await octokit.rest.users.getAuthenticated()
+      if (status < 400) {
+        return githubToken.token
+      }
+    } catch {}
```
Comment (erezrokah) [inline src/utils/init/config-github.js:41]: Should we add a log message telling the user their token is expired?

**Review by erezrokah — CHANGES_REQUESTED** [review]
This looks great @lukasholzer and I love the type definitions ❤️ 

Added two comments for your consideration

File: src/utils/gh-auth.js (hunk 1 above)
Comment (lukasholzer) [reply src/utils/gh-auth.js:125]: Thx have not known about the code guidelines 👍  will read through :)

File: src/utils/init/config-github.js (hunk 2 above)
Comment (lukasholzer) [reply src/utils/init/config-github.js:41]: Will do! Good point :)

File: src/utils/gh-auth.js (hunk 1 above)
Comment (erezrokah) [reply src/utils/gh-auth.js:125]: Yes, this should be a lint rule actually (e.g. the opposite of https://eslint.org/docs/rules/no-return-await)

Comment (erezrokah) [reply src/utils/gh-auth.js]: Update: I wonder why https://eslint.org/docs/rules/require-await didn't catch this. We have it [configured](https://github.com/netlify/eslint-config-node/blob/5dd59a59882b0a9cee04e376d3126a72da036daa/.eslintrc.js#L126)

File: src/utils/init/config-github.js [hunk 3]
```diff
@@ -193,7 +217,7 @@ module.exports = async function configGithub({ context, repoName, repoOwner, sit
   })
   await saveNetlifyToml({ repositoryRoot, config, configPath, baseDir, buildCmd, buildDir, functionsDir })
 
-  const octokit = getGitHubClient({ token })
+  const octokit = await getGitHubClient(token)
```
Comment (erezrokah) [inline src/utils/init/config-github.js:220]: Is this `await` needed (same for the tests)?

**Review by erezrokah — APPROVED** [review]
Added one more comment to clarify.

Looks 🚀

File: src/utils/init/config-github.js (hunk 3 above)
Comment (lukasholzer) [reply src/utils/init/config-github.js:220]: oh this is a old left over will change

Comment (lukasholzer) [reply src/utils/init/config-github.js:220]: What do you mean with the same for the tests? In the tests a different function is tested `getGitHubToken` this is actually async

Comment (erezrokah) [reply src/utils/init/config-github.js]: Ah, I missed that part (the fact it's async in the tests). You can ignore that part of my comment

**Review by erezrokah — APPROVED** [review]

**Pull request merged**